### PR TITLE
Test fix

### PR DESCRIPTION
**Pull request by veprbl**
This enables invocation of tests from setup.py, as well as their packaging for distribution at PyPI.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -26,10 +26,12 @@
     name='simanneal',
     version=version,
     description='Simulated Annealing in Python',
-    license='BSD',
+    license='ISC',
```
Comment (perrygeo) [inline setup.py]: Excellent catch, thanks.

File: setup.py [hunk 2]
```diff
@@ -26,10 +26,12 @@
...
     author='Matthew Perry',
     author_email='perrygeo@gmail.com',
     url='https://github.com/perrygeo/simanneal',
     long_description=LONG_DESCRIPTION,
-    packages=['simanneal'],
-    install_requires=[])
+    packages=['simanneal', 'tests'],
```
Comment (perrygeo) [inline setup.py]: This will "pollute" the global namespace with a package named `tests`. If we're going to include tests, we need to put tests as a submodule of the simanneal package.

File: tests/test_anneal.py [hunk 3]
```diff
@@ -65,18 +66,18 @@ def test_auto():
...
     # initial state, a randomly-ordered itinerary
     init_state = list(cities.keys())
     random.shuffle(init_state)
 
     tsp = TravellingSalesmanProblem(init_state, distance_matrix)
     tsp.copy_strategy = "slice"
 
-    statefile = str(tmpdir.join("state.pickle"))
-    tsp.save_state(fname=statefile)
+    with tempfile.NamedTemporaryFile('w') as statefile:
```
Comment (perrygeo) [inline tests/test_anneal.py]: 👍  - one question though:

 is this necessary to run tests with setup.py or ? Just trying to understand why this change is important.

Comment (veprbl) [reply tests/test_anneal.py]: Things seem to work without this when you use ```pytest```. Initially I didn't understand how the tests were supposed to be run, so I was trying to use ```nosetests```, which did not support this "magical" tmpdir argument. I can drop this change if you want.
